drivers/dac/ad5754r/ad5754r.c

```diff
@@ -0,0 +1,684 @@
+ *				dac channel address.
+ * @param reg_val - The data read from the register.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad5754r_read(struct ad5754r_dev *dev,
+		 uint8_t instr_addr,
+		 uint16_t *reg_val)
+{
+	int ret;
+	uint8_t buff[3];
+
+	if (!dev || !reg_val)
+		return -EINVAL;
+
+	/* Instruction frame format - {R/Wb, 0, R2, R1, R0, A2, A1, A0}
+	 * R2, R1, R0 : Register address bits
+	 * D2, D1, D0 : DAC address bits */
+	buff[0] = AD5754R_READ | instr_addr;
+
+	ret = no_os_spi_write_and_read(dev->spi_desc, buff, sizeof(buff));
+	if (ret)
+		return ret;
+
+	*reg_val = (buff[1] << 8) | buff[2];
+
+	return ret;
+}
+
+/**
+ * @brief Update specific register bits.
+ * @param dev - The device structure.
+ * @param instr_addr - The instruction address comprising register and
+ *				dac channel address.
+ * @param mask - Specific bits mask.
+ * @param reg_val - The data to be written.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad5754r_update_bits(struct ad5754r_dev *dev,
+			uint8_t instr_addr,
+			uint16_t mask,
+			uint16_t reg_val)
+{
+	int ret;
+	uint16_t reg_data;
+
+	if (!dev)
+		return -EINVAL;
+
+	ret = ad5754r_read(dev, instr_addr, &reg_data);
+	if (ret)
+		return ret;
+
+	reg_data &= ~mask;
+	reg_data |= reg_val;
+
+	return ad5754r_write(dev, instr_addr, reg_data);
+}
+
+/**
+ * @brief Read specific register bits using supplied mask.
+ * @param dev - The device structure.
+ * @param instr_addr - The instruction address comprising register and
+ *				dac channel address.
+ * @param mask - Specific bits mask.
+ * @param reg_val - The data read from the register.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad5754r_read_masked(struct ad5754r_dev *dev,
+			uint8_t instr_addr,
+			uint16_t mask,
+			uint16_t *reg_val)
+{
+	int ret;
+	uint16_t reg_data;
+
+	if (!dev)
+		return -EINVAL;
+
+	ret = ad5754r_read(dev, instr_addr, &reg_data);
+	if (ret)
+		return ret;
+
+	*reg_val = no_os_field_get(mask, reg_data);
+
+	return 0;
+}
+
+/**
+ * @brief Update DAC register value for specific channel.
+ * @param dev - The device structure.
+ * @param chn - The index of channel being addressed.
+ * @param value - The data to be written to the dac channel.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad5754r_update_dac_ch_register(struct ad5754r_dev *dev,
+				   uint8_t chn,
+				   uint16_t value)
+{
+	if (!dev || (chn > AD5754R_NUM_CHANNELS))
+		return -EINVAL;
+
+	return ad5754r_write(dev, AD5754R_PREPARE_INSTR_ADDR(AD5754R_REG_DAC,
+			     chn),
+			     value);
+}
+
+/**
+ * @brief Update DAC register value for all channels.
+ * @param dev - The device structure.
+ * @param value - The data to be written to the dac channel registers.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad5754r_update_dac_all_ch_registers(struct ad5754r_dev *dev,
+					uint16_t value)
+{
+	if (!dev)
+		return -EINVAL;
```

Comment:
The `ad5754r_write()` function already checks for this condition, so you can drop it here. The same applies in other places.

Reply:
I'm doing an additional LDAC toggle now. So the check is required for line 203 to not fail, before control gets to the write method